staging/src/k8s.io/apimachinery/pkg/util/proxy/transport.go

```diff
@@ -122,6 +122,16 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 
 var _ = net.RoundTripperWrapper(&Transport{})
 
+func (rt *Transport) CloseIdleConnections() {
```

Comment:
rather than making all round tripper layers implement this, would it make sense to implement a helper that calls WrappedRoundTripper() until it finds a round tripper that implements CloseIdleConnections()?

Reply:
util/net#DialerFor is an example of what that could look like